File: code/python/insertion_sort.py

def insertion_sort(arr):
    """
    Insertion sort algorithm.
    Args:
        arr: list of integers
    """
    for i in range(1, len(arr)):
        current = arr[i]
        for j in range(i-1, -1, -1):
            if arr[j] > current:
                arr[j+1] = arr[j]
            else:
                arr[j+1] = current
                break
            if j == 0:
                arr[j] = current


if __name__ == "__main__":
    arr = [5, 2, 4, 6, 1, 3]
    insertion_sort(arr)
    print(arr)

File: code/go/insertion_sort.go
package main

import "fmt"

func insertionSort(arr []int){
	for i := 1; i < len(arr); i++ {
		current := arr[i]
		for j := i - 1; j >= 0; j-- {
			if current < arr[j]{
				arr[j+1] = arr[j]
			} else {
				arr[j+1] = current
				break
			}
			if j == 0 {
				arr[j] = current
			}
		}
	}
}

func main(){
	arr := []int{5, 2, 4, 6, 1, 3}
	insertionSort(arr)
	fmt.Println(arr)
}